src/Magnum/Math/Bezier.h

```diff
@@ -0,0 +1,118 @@
+#ifndef Magnum_Math_Bezier_h
+#define Magnum_Math_Bezier_h
+
+/*
+    This file is part of Magnum.
+
+    Copyright © 2010, 2011, 2012, 2013, 2014, 2015, 2016
+              Vladimír Vondruš <mosra@centrum.cz>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+/** @file
+ * @brief Class @ref Magnum::Math::Bezier
+ */
+
+#include <array>
+#include "Vector.h"
+
+namespace Magnum { namespace Math {
+        /**
+        @brief Bezier
+        @tparam order       Order of Bezier curve
+        @tparam dimensions  Dimensions of the control points
+        @tparam T           Underlying data type
+
+        See <a href="https://en.wikipedia.org/wiki/B%C3%A9zier_curve">Bezier Curve</a>.
+        */
+        template<UnsignedInt order, UnsignedInt dimensions, class T> class Bezier {
+
+        public:
+
+            /** @brief Construct Bezier curve with the given array of control points */
+            explicit Bezier(const std::array<Vector<dimensions, T>, order+1> &points) {
```

Comment:
Okay, so thought about that a bit: what about using `Bezier<order, dimensions, T>` instead of `std::array<Vector<dimensions,T>, order + 1>` in the return type of `calculateIntermediatePoints()` so it would return `std::array<Bezier<order, dimensions, T>, order + 1>`? If you add `operator[]` accessor to the `Bezier` class, then it has basically the same API as `std::array` so it shouldn't be problem to use it and you also don't have to unpack it in any way to return the final `Bezier` instances.